src/basic/stat-util.c

```diff
@@ -151,6 +151,26 @@ int is_device_node(const char *path) {
         return verify_stat_at(AT_FDCWD, path, false, stat_verify_device_node, false);
 }
 
+int stat_verify_regular_or_block(const struct stat *st) {
+        assert(st);
+
+        if (S_ISDIR(st->st_mode))
+                return -EISDIR;
+
+        if (S_ISLNK(st->st_mode))
+                return -ELOOP;
+
+        if (!S_ISREG(st->st_mode) && !S_ISBLK(st->st_mode))
+                return -EBADFD;
```

Comment:
The errno chosen here is a bit weird because it's not something generically expressible.

Reply:
I'm trying to be consistent with what `stat_verify_regular` is doing, see https://github.com/systemd/systemd/pull/37209#discussion_r2054709417.